wtxmgr/tx_test.go

```diff
@@ -1364,3 +1364,205 @@ func TestRemoveUnminedTx(t *testing.T) {
 			len(unminedTxns))
 	}
 }
+
+// testInsertDoubleSpendTx is a helper test which double spends an output. The
```

Comment:
Excellent tests! 👍

Reply:
Maybe also add a test for the case where we have multiple unconfimed transactions stored, then a new mined tx comes in that invalidates all of them. 

For instance:
1. Spends outpoint A
2. Spends outpoint B

Now tx 3 comes in a  block and spends both A and B.

Reply:
Ended up modifying `TestInsertConfirmedDoubleSpendTx` to achieve this.